### ISSUE TITLE
"dnu publish --no-source" doesn't run on IIS Express

### ISSUE THREAD

**Issue by ChengTian** [aspnet/dnx#1523]
Reported by @Praburaj 

I verified `dnu publish` with:
Runtime: dnx-coreclr-win-x86.1.0.0-beta4-11390 (latest release branch)
MusicStore: latest release branch
AspNet.Loader: 1.0.0-beta5-11379

Running `web.cmd` works well. However, running the bundled app with IIS Express gives:
![capture](https://cloud.githubusercontent.com/assets/1383883/6878921/6294f040-d4a4-11e4-8fc8-766242942999.PNG)

@muratg @davidfowl , this bug is in release branch and it should be a beta4 issue.

**Comment by ChengTian** [comment]
It also happens with CLR. I tested with dnx-clr-win-x86.1.0.0-beta4-11390 (latest release branch). `web.cmd` works fine, IIS Express gives:
![capture](https://cloud.githubusercontent.com/assets/1383883/6878995/53e9a192-d4a6-11e4-961f-d608c3943f55.PNG)

**Comment by davidfowl** [comment]
Odd.

**Comment by davidfowl** [comment]
Where did this repro? The music store tests or by hand?

**Comment by ChengTian** [comment]
By hand. We don't have coverage for dnu publish in MusicStore. @Praburaj just found the root cause of a blocking issue (https://github.com/aspnet/dnx/issues/1515) and he is going to add coverage for dun publish.

**Comment by ChengTian** [comment]
Reported by @kirthik 
Need to verify `dnu publish` (without `--no-source`) and deploy to IIS

**Comment by ChengTian** [comment]
@kirthik , if you see `Could not load file or assembly 'dnx.clr.managed' or one of its dependencies`, it's by design. You should either `dnu publish` with x64 runtime, or turn on `Enable 32-Bit Applications` for your application pool.

**Comment by kirthik** [comment]
There is nothing in the error message that tells the user that. How would a user figure it out?

**Comment by ChengTian** [comment]
@kirthik , if you follow instructions on the error page and turn on assembly binding logging, you can figure it out.

@davidfowl , do we want to give more friendly error message when a architecture-mismatch is detected in `AspNet.Loader`? (For example, when a x86 runtime is needed but there is only a x64 runtime bundled)

**Comment by davidfowl** [comment]
That would be great!

**Comment by ChengTian** [comment]
Filed an issue for more friendly error message: https://github.com/aspnet/Helios/issues/127

**Comment by muratg** [comment]
Closing this one as this is by design.  Error message improvements are tracked in the Helios bug 127

**Comment by dg9ngf** [comment]
The other project/issue you referenced is gone. Where can I find it? Just stumbled across this error and this page helped me to fix it.

**Comment by muratg** [comment]
@dg9ngf That's because that's in a currently internal project on GitHub.  However, the issue is fixed in the dev branch already, and you'll be getting a `DirectoryNotFoundException` in this case.  Feel free to test it with the beta5 bits.

**Comment by fekberg** [comment]
I had the same issue and as per @ChengTian instructions, changing `Enable 32-Bit Applications` to `true` solved it.

**Comment by stodolos** [comment]
I have come across this issue and tried to switch to x64, however, if I launch with IIS Express, it is stuck on the x86 version of beta8.  I've tried many things suggested on the web, but not matter what it seems to be stuck on x86.  If I launch the web command, it launches DNX in x64 mode and everything works fine, but then I am not sure how to set the same routing rules that I have in my web.config that IIS uses.  I would prefer IIS Express to do what I'm asking either way :)

**Comment by stodolos** [comment]
I take that back... it *might* be running the DNX in x64 (as I didn't see the *32 next to the task in the task manager), but it still says the following in the error:

"LOG: Appbase = file:///C:/Users/CSpanellis/.dnx/runtimes/dnx-clr-win-x86.1.0.0-beta8/bin
LOG: Initial PrivatePath = NULL"

**Comment by edmccaffrey** [comment]
I'm seeing this error on an Azure standard size instance. The Azure instance is set to 32-bit in application settings, and the publish settings is dnx-clr-win-x86.

**Comment by stodolos** [comment]
I actually figured out my issue...  I had set this config setting a while back when I was first configuring my Angular set up:

<modules runAllManagedModulesForAllRequests="true" />

I narrowed it down to that after seeing as how launching it outside of IIS Express worked fine (aside from my routing settings in my config file), so I then looked at my config file as a possible culprit.  I then started commenting out sections that were different than the standard config, and by process of elimination eventually narrowed it down to this line.

All is good now for my set up.
